x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -71,11 +71,16 @@ export interface ActionConnectorTableItem extends ActionConnector {
   actionType: ActionType['name'];
 }
 
+export interface AlertTypeActionVariables {
+  context: Record<string, string>;
+  state: Record<string, string>;
+}
+
 export interface AlertType {
   id: string;
   name: string;
   actionGroups: ActionGroup[];
-  actionVariables: string[];
+  actionVariables: AlertTypeActionVariables;
```

Comment:
I think it makes sense to have it as optional and do not show the empty dropdown list in the case it is undefined

Reply:
messageVariables which use actionVariables in UI is already optional for the ActionParamsProps as well

Reply:
The action variables list will never be empty by the time the context menu displays, as there are 5-6 "always available" variables that come from the alert library; eg, `alertName`.

I've also reshaped these since you reviewed, as the shape was radically different than `actionGroups` (which is an array of id / name object), so I changed the actionVariables to match.  This also means we won't have prototype pollution issues (even if not our fault), as we won't be using unknown properties on objects.